client/my-sites/plans/ecommerce-trial/ecommerce-trial-banner/index.tsx

```diff
@@ -41,7 +41,8 @@ const ECommerceTrialBanner = ( props: ECommerceTrialBannerProps ) => {
 	/**
 	 * Trial progress from 0 to 1
 	 */
-	const trialProgress = 1 - eCommerceTrialDaysLeft / trialDuration;
+	const trialProgress =
+		eCommerceTrialDaysLeft <= trialDuration ? 1 - eCommerceTrialDaysLeft / trialDuration : 0;
```

Comment:
Two thoughts:
1. This is really complicated to read! Could we use `Math.max()` to simplify the expression?
2. I was testing with a RTL language earlier today (I think it was Hebrew) and things looked a bit weird -- it may be worth looking at that as well while you're looking at this component.
```suggestion
	// Protect against values < 0 by wrapping the code in Math.max()
	const trialProgress = Math.max( 1 - eCommerceTrialDaysLeft / trialDuration, 0 );
```

Reply:
I created this issue with the problems I found: https://github.com/Automattic/wp-calypso/issues/73392